### ISSUE TITLE
RuleTester swallows parsing errors when fixing

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#4779]
I came across a problem with `RuleTester` where it swallows parsing errors when there's `output` specified in an invalid test case. Example:

```js
{
    code: "var foo = `bar`;",
    output: "var foo = 'bar';",
    options: ["single"],
    errors: [{ message: "Strings must use singlequote.", type: "Literal"}]
},
```

This will fail to parse because `ecmaVersion` isn't set to 6, however, `RuleTester` just outputs "Output does not match." It should output the parsing error (I spent about an hour tracking this down).
